src/System.Text.Json/porting_guide/README.md

```diff
@@ -0,0 +1,173 @@
+  - Code sample for async writing to a `Stream` / `PipeWriter` - TBD
+
+```C#
+// Assume ArrayBufferWriter is a class that implements IBufferWriter<byte> and has a CopyTo(Stream) method
+static void WriteJson(ArrayBufferWriter output, Stream stream)
+{
+    const int SyncWriteThreshold = 1_000; // Copy to stream after writing this many bytes
+
+    var json = new Utf8JsonWriter(output, state: default);
+
+    long prevBytesWritten = 0;
+
+    // Write some JSON, let's say an array of JSON objects in a loop
+    json.WriteStartArray();
+
+    for (int j = 0; j < numberOfElements; j++)
+    {
+        // json.Write...
+
+        prevBytesWritten = json.BytesWritten - prevBytesWritten;
+        if (prevBytesWritten > SyncWriteThreshold)
+        {
+            json.Flush(isFinalBlock: false);
+            output.CopyTo(stream);
+        }
+    }
+
+    json.WriteEndArray();
+    json.Flush(isFinalBlock: true);
+    output.CopyTo(stream);
+}
+```
+
+### Use Text Encoded as UTF-8 for Best Performance
+
+* As the type names suggest, if you are looking for the best possible performance while using the `Utf8JsonReader` and `Utf8JsonWriter`, try to read/write JSON payloads already encoded as UTF-8 text (rather than UTF-16 strings). For example, if you are writing string literals, consider caching them as static byte arrays, and write those instead.
+
+```C#
+private const string PropertyName = "name";
+private static readonly byte[] PropertyNameBytes = Encoding.UTF8.GetBytes(PropertyName);
+// ...
+jsonWriter.WriteNumber(PropertyNameBytes, 42, escape: false);
+// ...
+```
+
+### Using a Read-only JsonDocument
+
+* `JsonDocument` provides the ability to parse JSON data and build a **read-only** Document Object Model (DOM) with low allocations for common payload sizes (i.e. < 1 MB). It does this by building an in-memory view of the data into a pooled buffer. Therefore, unlike `JObject`/`JArray`, this type is `IDisposable` and needs to be used inside a using block. Additionally, since the DOM is read-only, it doesn't provide the ability to add/remove JSON elements. If your scenario requires a writable DOM or if you need to build up a JSON payload from scratch, continue to use `Newtonsoft.Json` and `JObject`/`JArray` since that is currently an unsupported feature in the new stack.
```

Comment:
Change "please continue to use Newtonsoft" to "please open a bug and tell us about your scenario; in the mean time continue to use Newtonsoft.."